### PR TITLE
Add CI with buildifier, clang-format, clang-tidy, and bazel build/test.

### PR DESCRIPTION
**Pull request by benjmarks22**

### CODE REVIEW

File: src/server/greeter_server_main.cc [hunk 1]
```diff
@@ -1,16 +1,13 @@
-
-#include <string>
-#include <memory>
 #include <iostream>
+#include <memory>
+#include <string>
 
 #include "grpc/grpc.h"
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/server/greeter_server_main.cc:5]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/server/greeter_server.h [hunk 2]
```diff
@@ -3,18 +3,17 @@
 #define SRC_SERVER_GREETER_SERVER_H_
 
 #include "grpcpp/server_context.h"
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/server/greeter_server.h:5]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/server/greeter_server_test.cc [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#include "src/server/greeter_server.h"
+
+#include "gmock/gmock.h"
+#include "grpcpp/server_context.h"
+#include "gtest/gtest.h"
+
+namespace {
+TEST(GreeterServerTest, ResponseIsCorrect) {
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/server/greeter_server_test.cc:8]: It was due to a bug with compiling external sources. That bug is now fixed.

[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/server/greeter_server_test.cc:8]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/client/greeter_client.h [hunk 4]
```diff
@@ -6,21 +6,21 @@
 
 #include "grpcpp/channel.h"
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/client/greeter_client.h:7]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/client/greeter_client.h [hunk 5]
```diff
@@ -6,21 +6,21 @@
...
 
 namespace client {
 
 class Client {
  public:
-  Client(std::shared_ptr<grpc::Channel> channel)
+  explicit Client(std::shared_ptr<grpc::Channel> channel)
       : stub_(greeter::Greeter::NewStub(channel)) {}
 
   void Greet();
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/client/greeter_client.h:18]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/client/greeter_client.cc [hunk 6]
```diff
@@ -1,22 +1,21 @@
...
-    } else {
-      std::cout << "Greet rpc failed " << status.error_message() << std::endl;
-    }
+void Client::Greet() {
+  greeter::GreetRequest request;
+  greeter::GreetResponse response;
+  grpc::ClientContext context;
+  request.set_name("Test");
+  grpc::Status status = stub_->Greet(&context, request, &response);
+  if (status.ok()) {
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/client/greeter_client.cc:14]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/client/greeter_client_main.cc [hunk 7]
```diff
@@ -3,14 +3,12 @@
 #include "grpc/grpc.h"
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply src/client/greeter_client_main.cc:3]: It was due to a bug with compiling external sources. That bug is now fixed.

File: src/server/greeter_server_test.cc [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  grpc::ServerContext context;
+  greeter::GreetRequest request;
+  greeter::GreetResponse response;
+  request.set_name("Test");
+  EXPECT_TRUE(
```
Comment (n-triple-a) [inline src/server/greeter_server_test.cc:13]: nit: is EXPECT_OK open sourced? If yes, I guess that's preferred.

**Review by n-triple-a — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -1,2 +1,57 @@
...
+To run the server, run:
+
+`bazel run //src/server:greeter_server_main`
+
+### Client
+
+To run the client, run:
+
+`bazel run //src/client:greeter_client_main`
+
```
[the remark this answers is not in the document]
Comment (benjmarks22) [reply README.md:24]: Done.

**Pull request merged**